backend/api/login/views.py

```diff
@@ -15,69 +15,81 @@
 logger = logging.getLogger('backend')
 
 
-class ClientRenderedException(Exception):
+class LoginException(Exception):
     pass
 
 
 class LoginRequest(APIView):
     parser_classes = (MultiPartParser, FormParser, JSONParser,)
 
     # For debugging purpose :
-    #   - Return sugang_list at success
+    #   - Return taken_list at success
     def post(self, request, option):
         try:
             if option == 'mysnu':
                 logger.debug('mySNU login request with sessionid: ' + str(request.session.session_key))
+
                 user_id = request.data.get('user_id', 'nid')
                 password = request.data.get('password', 'npw')
-                sugang_list = mysnu.crawl_credit(user_id, password)
-                if sugang_list is None:
-                    raise ClientRenderedException('로그인에 실패했습니다.')
-            else:  # option == 'file':
+
+                taken_list = mysnu.crawl_credit(user_id, password)
+
+                if taken_list is None:
+                    raise LoginException('로그인에 실패했습니다.')
+
+            elif option == 'file':
                 logger.debug('File login request with sessionid: ' + str(request.session.session_key))
                 request.data['major_info'] = True
                 filename = request.data.get('filename', 'nofile')
                 file = request.data.get(filename, None)
                 if file is None:
-                    raise ClientRenderedException('파일이 올바르지 않습니다.')
-                text = file.read()
+                    raise LoginException('파일이 올바르지 않습니다.')
+                text = file.read().decode('euc-kr', errors='ignore')
                 file.close()
-                text = text.decode('euc-kr', errors='ignore')
-                sugang_list = parse_credit(text)
+                taken_list = parse_credit(text)
```

Comment:
I think just adding `try: catch Exception:` here will be great.
Catching exceptions in `parse_credit` will be very difficult.